### ISSUE TITLE
BUG: Singularity build fails

### ISSUE THREAD

**Issue by JWDebler** [ccdmb/predector#34]
**Describe the bug**
Build crashes using singularity tag

**To Reproduce**
```
 curl -s "https://raw.githubusercontent.com/ccdmb/predector/master/install.sh" | bash -s singularity -3 signalp-3.0.Linux.tar.Z     -4 signalp-4.1g.Linux.tar.gz     -5 signalp-5.0b.Linux.tar.gz     -t targetp-2.0.Linux.tar.gz     -d deeploc-1.0.All.tar.gz     -m tmhmm-2.0c.Linux.tar.gz  -p phobius101_linux.tar.gz
```

**Screenshots or copy-pasted terminal outut **
```
INFO:    Starting build...
Getting image source signatures
Copying blob sha256:68ced04f60ab5c7a5f1d0b0b4e7572c5a4c8cce44866513d30d9df1a15277d6b
 25.84 MiB / 25.84 MiB [====================================================] 0s
Copying blob sha256:9c388eb6d33c40662539172f0d9a357287bd1cd171692ca5c08db2886bc527c3
 76.50 MiB / 76.50 MiB [====================================================] 1s
Copying blob sha256:96cf53b3a9dd496f4c91ab86eeadca2c8a31210c2e5c2a82badbb0dcf8c8f76b
 48.03 MiB / 48.03 MiB [====================================================] 0s
Copying blob sha256:4c652f7bbbd31c7bdc6db188af3e21e51ba78770396d03ad1b29c193bb334328
 13.56 MiB / 13.56 MiB [====================================================] 1s
Copying blob sha256:cceea6a4aa09dc45cf08aae7f951ac4a96d3f79dd27473c0645f714b2097d193
 429 B / 429 B [============================================================] 0s
Copying blob sha256:d6452039e513df89b320ac1063567b8f1eb9f728b776f2596dcc71e14aeca466
 898.51 MiB / 898.51 MiB [================================================] 2m4s
Copying config sha256:fba358169658187e2332dfaa8d7aac5d4d91878647c487f971a12f452ba941bc
 5.72 KiB / 5.72 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures

INFO:    Running setup scriptlet
+ set +x
singularity image-build: relocation error: /lib/x86_64-linux-gnu/libnss_files.so.2: symbol __libc_readline_unlocked version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
FATAL:   While performing build: while running engine: exit status 127

ERROR: creating the singularity image failed
```

**Operating system (please enter the following information as appropriate):**
 - OS/Linux distribution: Ubuntu 18.04.4 LTS
 - Dependency management: Singularity
 - Nimbus VM

**Comment by darcyabjones** [comment]
This is a bit weird.

It looks like it might be more to do with the singularity installation or the base image for the container.

See:
https://github.com/hpcng/singularity/issues/3917

I'll look at this when I have some more time.
But it might be hard to debug and fix in the more general case.
 
Thanks for letting me know.

**Comment by JWDebler** [comment]
I converted the local docker image to singularity with `singularity build predector.sif docker-daemon://predector/predector:0.1.0-dev` and then ran `nextflow run -profile test -with-singularity predector.sif -resume ccdmb/predector`. That crashes with:
```
Error executing process > 'extract_effector_seqs'

Caused by:
  Process `extract_effector_seqs` terminated with an error exit status (1)

Command executed:

  tail -n+2 effectors.tsv     | awk -F' ' '{printf(">%s\n%s\n", $4, $12)}'     > effectors.fasta                                                                                                               
Command exit status:
  1

Command output:
  (empty)                                                                                                                                                                                                      
Command error:
  tail: cannot open 'effectors.tsv' for reading: No such file or directory
```
I then built another singularity image via `sudo -E singularity build predector_built.sif ./singularity.def` and that leads to the original error of this issue.

**Comment by darcyabjones** [comment]
I wasn't able to replicate this in my VM.
Sorry.

How did you install Singularity?
Do you know what version it was?

Also, do you have any `conda` environments activated when you try to run the build commands?

Thanks.

**Comment by JWDebler** [comment]
Not sure. But I am struggling reinstalling it at the moment, because it keeps claiming go isn't installed when it certainly is. I'll keep trying :-)

**Comment by darcyabjones** [comment]
Installing the Go compiler is a bit of a weird process.
I find installing from their [binaries](https://golang.org/doc/install) and setting the `PATH` and `GOPATH`  is the most reliable way.

Don't know how you installed it but i've found the Go distributions installed by system package managers (e.g. apt) don't always work properly.

**Comment by darcyabjones** [comment]
Don't stress too much if you can't get it going.
I super appreciate all of the issues and bugs you've found, and help you've given already and I assume you've got other things to do.
Not to discourage you from doing more, just saying don't feel obliged.

I'll see if I can debug this on a few different operating systems when I have a bit more time.

**Comment by JWDebler** [comment]
I'm happy to keep testing :-)

Managed to reinstall GO (had to remove /usr/local/go) for it to properly install a new version, which then let me install Singularity (3.5.2). 
I now managed to build the singularity container, but the testrun failed:

`nextflow run -profile test -with-singularity predector.sif -r 0.1.0-dev ccdmb/predector -latest`

```
Error executing process > 'extract_effector_seqs'

Caused by:
  Process `extract_effector_seqs` terminated with an error exit status (1)

Command executed:

  tail -n+2 effectors.tsv     | awk -F'u' '{printf(">%s\n%s\n", $4, $12)}'     > effectors.fasta

Command exit status:
  1

Command output:
  (empty)

Command error:
  tail: cannot open 'effectors.tsv' for reading: No such file or directory
```

**Comment by darcyabjones** [comment]
This one I can reproduce, at first I thought this would be resolved with the `-latest` or `-r 0.1.0-dev` option.
It looks like I missed a config option for singularity.
See if this works now.

```
nextflow run -profile test -with-singularity predector.sif -r 3b9d1ef ccdmb/predector
```

**Comment by JWDebler** [comment]
Thanks Darcy, I had to add the `-latest` flag to make it work, see below. But otherwise this seems to be fixed :-)

```
(base) ubuntu@bifo-server:~/predector$ nextflow run -profile test -with-singularity predector.sif -r 3b9d1ef16a4ce36d871f255ff40877acc3fc95a7 ccdmb/predector
N E X T F L O W  ~  version 20.01.0
Unknown error accessing project `ccdmb/predector` -- Repository may be corrupted: /home/ubuntu/.nextflow/assets/ccdmb/predector

(base) ubuntu@bifo-server:~/predector$ nextflow run -profile test -with-singularity predector.sif -r 3b9d1ef16a4ce36d871f255ff40877acc3fc95a7 ccdmb/predector -latest
N E X T F L O W  ~  version 20.01.0        
Pulling ccdmb/predector ...
Launching `ccdmb/predector` [adoring_borg] - revision: 3b9d1ef16a4ce36d871f255ff40877acc3fc95a7                      
```
